ansible/files/bootstrap/consul/post/task/vault-integration.yml

```diff
@@ -1,79 +1,136 @@
+#- name: Vault - enable vault's consul secrets engine
+#  shell: vault secrets enable consul
+#  environment:
+#    VAULT_TOKEN: "{{ vault_master_token }}"
+
+- name: "{{ service }} - create management token for Vault, with global-management policy attached to it"
   shell:
     cmd: consul acl token create -policy-name=global-management -format=json | jq -r .SecretID
+  register: secret_id
   environment:
     CONSUL_HTTP_TOKEN: "{{ lookup('env', 'consul_master_token') }}"
-  register: secret_id
 
-- name: "{{ service }} - post/vault-integration - configure Vault to connect and authenticate to Consul"
+- name: "{{ service }} - configure Vault to connect and authenticate to Consul"
   shell:
-    cmd: vault write consul/config/access address=127.0.0.1:8500 token="{{ secret_id.stdout }}"
+    cmd: vault write consul/config/access address="{{ consul_host }}" token="{{ secret_id.stdout }}"
   environment:
     VAULT_TOKEN: "{{ lookup('env', 'vault_master_token') }}"
 
-# example 1
-# 1. Create consul policy for admin team
+# example 1 - Admin policy, admin role
+# 1. Create consul policy for admin
 # 2. Map role to policy
 # 3. Generate token(s)
-- name: "{{ service }} - post/vault-integration - create admin team policy"
-  shell:
-    cmd: consul acl policy create -name admin-team-policy -rules @/etc/ansible/templates/conf/consul/policies/admin_team_policy.hcl
+- name: "{{ service }} - create admin team policy"
+  shell: consul acl policy create -name {{ consul.policy.admin }} -rules @/etc/ansible/templates/conf/consul/policies/admin_policy.hcl
   environment:
     CONSUL_HTTP_TOKEN: "{{ lookup('env', 'consul_master_token') }}"
 
-- name: "{{ service }} - post/vault-integration - map admin-team role with admin-team-policy to a Consul ACL policy"
+- name: "{{ service }} - map admin role with admin-policy to a Consul ACL policy"
   shell:
-    cmd: vault write consul/roles/admin-team policies=admin-team-policy
+    cmd: vault write consul/roles/admin-role policies="{{ consul.policy.admin }}"
+  register: admin_role_policy_register
   environment:
     VAULT_TOKEN: "{{ lookup('env', 'vault_master_token') }}"
 
-- name: "{{ service }} - post/vault-integration - example, generate token for admin-team member"
+- name: "{{ service }} - print admin team policy register"
+  debug:
+    msg: "{{ admin_role_policy_register }}"
+
+- name: "{{ service }} - example, generate token for role admin"
   shell:
-    cmd: vault read consul/creds/admin-team -format=json | jq -r .data.token
-  register: admin_team_token
+    cmd: vault read consul/creds/admin-role -format=json | jq -r .data.token
+  register: admin_role_token
   environment:
     VAULT_TOKEN: "{{ lookup('env', 'vault_master_token') }}"
 
-- name: "{{ service }} - post/vault-integration - Print admin team token"
+- name: "{{ service }} - print admin role token"
   debug:
-    msg: "{{ admin_team_token.stdout }}"
+    msg: "{{ admin_role_token }}"
```

Comment:
Do we need to include debugs?

Reply:
I will remove it

Reply:
fixed https://github.com/fredrikhgrelland/vagrant-hashistack/pull/414/commits/ba0ad9a379730ef81b230690e458cb31c16d82b8